src/main/java/twilightforest/world/components/TFCavesCarver.java

```diff
@@ -78,6 +77,8 @@ public boolean carve(CarvingContext ctx, CaveCarverConfiguration config, ChunkAc
 
 	@Override
 	protected boolean carveBlock(CarvingContext ctx, CaveCarverConfiguration config, ChunkAccess access, Function<BlockPos, Biome> biomePos, CarvingMask mask, BlockPos.MutableBlockPos pos, BlockPos.MutableBlockPos posUp, Aquifer aquifer, MutableBoolean isSurface) {
+		//TODO mayby another way...?
+		Random random = new Random();
```

Comment:
This is terrible for deterministic world generation, please see if you can get an instance of a `Random` from one of the parameters or seed it instead